core/src/main/java/com/sequenceiq/cloudbreak/service/stack/StackApiViewService.java

```diff
@@ -50,10 +58,26 @@ public StackApiView save(StackApiView stackApiView) {
         return stackApiViewRepository.save(stackApiView);
     }
 
-    public Set<StackViewResponse> retrieveStackViewsByWorkspaceId(Long workspaceId) {
+    public Set<StackViewResponse> retrieveStackViewsByWorkspaceId(Long workspaceId, String environmentName, boolean dataLakeOnly) {
         try {
-            return transactionService.required(() ->
-                    convertStackViews(stackApiViewRepository.findByWorkspaceId(workspaceId)));
+            Set<StackViewResponse> stackViewResponses;
+            if (StringUtils.isEmpty(environmentName)) {
+                stackViewResponses = transactionService.required(() ->
```

Comment:
please check whether the transanctionService is necessary here, and if not, please remove it.

Reply:
lazy init exceptions are the reason